### PR TITLE
Paper cash stacks visually with it's value.

### LINKED ISSUE #31928: Space cash is awful to sort and kind of buggy.
>Blue cash equals 200 credits
>There are stacks of 5 and 20, worth 1000 and 4000 respectively, still using the blue sprite.
>Both are apparently worth 1000 and two hundred at the same time
>Is actually worth 5000 in total.

What the hell??

We have stack icon changes now, make it so it actually changes sprites based on how much cash there is, and tell us exactly how much there is in there rather than how many chunks of space money there are. I don't know, limit it to stacks of 5,000 cash with the blood red 1k cash icon?

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

image
_The top row is single bills of all the various values, while the bottom row is stacks with the same value, made of multiple smaller bills._
I've changed the update_icon_state proc for paper money, so that multiple bills when combined, will visually display their icon associated with the closest paper bill in value, similar to how holocredits update their icon based on value ranges. If nothing else, it improves the clarity of how much a stack of bills is worth, if you even manage to get enough paper bills.

## Why It's Good For The Game

Resolves #31928 considering that space cash is more of a tertiary source of money in the current system anyway, but at least the visual clarity when using it is improved for player use.

## Changelog
:cl:
fix: Paper cash, when stacked, now displays an icon scaling with it's appropriate value.
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/cash.dm [hunk 1]
```diff
@@ -20,19 +20,44 @@
...
 	return (amount*value)
 
 /obj/item/stack/spacecash/merge(obj/item/stack/S)
 	. = ..()
+	update_icon_state()
 	update_desc()
 
 /obj/item/stack/spacecash/use(used, transfer = FALSE)
 	. = ..()
+	update_icon_state()
```
Comment (ShizCalev) [inline code/game/objects/items/stacks/cash.dm:35]: the parent proc already does this

File: code/game/objects/items/stacks/cash.dm [hunk 2]
```diff
@@ -20,19 +20,44 @@
...
 	var/total_worth = get_item_credit_value()
-	desc = "It's worth [total_worth] credit[( total_worth > 1 ) ? "s" : ""]"
+	desc = "It's worth [total_worth] credit[( total_worth > 1 ) ? "s" : ""] each."
 
 /obj/item/stack/spacecash/get_item_credit_value()
 	return (amount*value)
 
 /obj/item/stack/spacecash/merge(obj/item/stack/S)
 	. = ..()
+	update_icon_state()
```
Comment (ShizCalev) [inline code/game/objects/items/stacks/cash.dm:30]: parent already does this.

File: code/game/objects/items/stacks/cash.dm [hunk 3]
```diff
@@ -20,19 +20,44 @@
 
 /obj/item/stack/spacecash/proc/update_desc()
 	var/total_worth = get_item_credit_value()
-	desc = "It's worth [total_worth] credit[( total_worth > 1 ) ? "s" : ""]"
+	desc = "It's worth [total_worth] credit[( total_worth > 1 ) ? "s" : ""] each."
```
Comment (ShizCalev) [inline code/game/objects/items/stacks/cash.dm:23]: >stack of 40 $10 bills
>it's worth 400 credits each.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/cash.dm [hunk 4]
```diff
@@ -20,19 +20,44 @@
...
+	update_icon_state()
 	update_desc()
 
 /obj/item/stack/spacecash/use(used, transfer = FALSE)
 	. = ..()
+	update_icon_state()
 	update_desc()
 
+/obj/item/stack/spacecash/update_icon_state()
+	. = ..()
```
Comment (ShizCalev) [inline code/game/objects/items/stacks/cash.dm:39]: why are you calling the parent if you're overriding the results anyway?

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Review by ShizCalev — APPROVED** [review]

**Comment by Dorsisdwarf** [comment]
is there supposed to be a visual difference between top and bottom row? the sprites all look identical

**Comment by ShizCalev** [comment]
@Dorsisdwarf they're the same sprite.

**Pull request merged**